lib/memory_profiler/results.rb

```diff
@@ -47,16 +47,23 @@ def register_results(allocated, retained, top)
     end
 
     def string_report(data, top)
-      data.values
-          .keep_if { |stat| stat.string_value }
-          .map! { |stat| [stat.string_value, stat.location] }
-          .group_by { |string, _location| string }
-          .sort_by {|string, list| [-list.size, string] }
-          .first(top)
-          .map { |string, list| [string, list.group_by { |_string, location| location }
-                                             .map { |location, locations| [location, locations.size] }
-                                ]
-          }
+      grouped_strings = data.values.
+        keep_if { |stat| stat.string_value }.
+        group_by { |stat| stat.string_value.object_id }.
+        values
```

Comment:
Was trying to find a way to reduce the number of `.values` calls, but couldn't do it in a way that was better than what we have.  Most of my solutions may have reduced the number of cycles through the array, but did so a making the blocks for those more complex (using more ruby), which meant more time evaluating ruby code and less time in the underlying c code these methods are implemented in.

The other (wrong) optimization I considered was flipping the first `.values` call with `.keep_if`:

```ruby
grouped_strings = data.keep_if { |stat| stat.string_value }.
  values.
  # ...
```

But that would then mutate the original hash, which wouldn't be good.  Anyway, mostly sharing to let you know what I was trying to do, but couldn't, for reference later.